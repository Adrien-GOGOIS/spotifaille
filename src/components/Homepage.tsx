import { FunctionComponent, useEffect, useState } from 'react';
import Favorites from './UserFavorites/Favorites';

import { authentificationServiceInstance } from '../services/authentification.service';
import { favoriteServiceInstance } from '../services/favorite.service';
import Title from './Title';

const Homepage: FunctionComponent  = () => {
	const [ isLoggedIn, setLoggedIn ] = useState<boolean>(false);
	const access_token = favoriteServiceInstance.getAccessToken();

	useEffect(() => {
		if (access_token) {
			setLoggedIn(true);
		}
	}, []);

	return (
		<div>
			<Title />
			{ !isLoggedIn &&
				<button
					onClick={() => window.open(authentificationServiceInstance.getAuthorizeHref(), '_self')}
					className='btn border-2 p-2'
				>
					Connectez-vous Ã  Spotify
				</button> 
			}
			{ isLoggedIn && <Favorites/> }
		</div>
	);
}

export default Homepage;