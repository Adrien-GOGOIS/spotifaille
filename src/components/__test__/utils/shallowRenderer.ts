/* eslint-disable @typescript-eslint/no-magic-numbers */
/* eslint-disable @typescript-eslint/no-type-alias */
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = ShallowRenderer.createRenderer();

type Input = Parameters<typeof renderer.render>[0];
type Output = ReturnType<typeof renderer.getRenderOutput>;

export const shallowRender = (node: Input): Output => {
  renderer.render(node);
  return renderer.getRenderOutput();
};
