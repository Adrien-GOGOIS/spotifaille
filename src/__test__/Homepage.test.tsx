import Homepage from '../components/Homepage';

import { shallowRender } from './utils/shallowRenderer';

describe('Homepage Component', () => {
  describe('Rendering', () => {
    test('should display component', () => {
      // eslint-disable-next-line testing-library/render-result-naming-convention
      const homepage = shallowRender(<Homepage />);

      expect(homepage).toMatchSnapshot();
    });
  });
});