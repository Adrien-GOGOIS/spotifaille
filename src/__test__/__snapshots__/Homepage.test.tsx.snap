// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupFormComponent Rendering should display component 1`] = `
<div>
  <button
    className="btn border-2 p-2"
    onClick={[Function]}
  >
    Connectez-vous à Spotify
  </button>
</div>
`;

exports[`Homepage Component Rendering should display component 1`] = `
<div>
  <button
    className="btn border-2 p-2"
    onClick={[Function]}
  >
    Connectez-vous à Spotify
  </button>
</div>
`;
